File: server/src/main/java/com/victovito/chessapp/models/AvailableCastlesModel.java
package com.victovito.chessapp.models;

import com.victovito.chessapp.objects.AvailableCastles;

public class AvailableCastlesModel {
    boolean kingSide;
    boolean queenSide;

    public AvailableCastlesModel() {}

    public AvailableCastlesModel(AvailableCastles availableCastles) {
        kingSide = availableCastles.getKingSide();
        queenSide = availableCastles.getQueenSide();
    }

    public boolean getKingSide() {
        return kingSide;
    }

    public boolean getQueenSide() {
        return queenSide;
    }
    
}

File: client/src/models/AvailableCastlesModel.js
export default class AvailableCastlesModel {
    /** @type {boolean} */
    kingSide;
    /** @type {boolean} */
    queenSide;

    /**
     * @param {AvailableCastlesModel} data 
     */
    constructor(data) {
        this.kingSide = data.kingSide;
        this.queenSide = data.queenSide;
    }
}